conformance/report.go

```diff
@@ -55,33 +92,50 @@ var _ = ReportAfterSuite("MCS conformance report", func(report Report) {
 			continue
 		}
 
-		for _, label := range specReport.Labels() {
+		for _, label := range reportingLabels {
+			if !slices.Contains(specReport.Labels(), label) {
+				continue
+			}
+
 			if testGroupMap[label] == nil {
 				testGroupMap[label] = &testGrouping{
 					Name: label,
 				}
 			}
 
-			ref := ""
+			info := testInfo{
+				Desc:       specReport.FullText(),
+				Conformant: true,
+			}
+
 			for i := range specReport.ReportEntries {
 				if specReport.ReportEntries[i].Name == SpecRefReportEntry {
-					ref = specReport.ReportEntries[i].GetRawValue().(string)
-					break
+					info.Ref = specReport.ReportEntries[i].GetRawValue().(string)
+				} else if specReport.ReportEntries[i].Name == NonConformantReportEntry {
+					info.Conformant = false
+					info.Message = specReport.ReportEntries[i].GetRawValue().(string)
 				}
 			}
 
-			testGroupMap[label].Tests = append(testGroupMap[label].Tests, testInfo{
-				Desc: specReport.FullText(),
-				Ref:  ref,
-				Pass: specReport.State == types.SpecStatePassed,
-			})
+			// If the spec failed (ie didn't pass) not due to non-conformance then we assume it encountered
+			// an unexpected error preventing conformance from being determined, and thus we'll report the
+			// conformance status as unknown.
+			if specReport.State != types.SpecStatePassed && info.Conformant {
+				info.Failed = true
+				info.Message = parseFailureMessage(specReport.FailureMessage())
```

Comment:
I wonder if we could generalize `parseFailureMessage` to be `parseSpecReportMessage` (or something), to populate `info.Message` in either the non-conformant or failure case. That way we can drop this:

```
if info.Message != "" {
    info.Message = " - " + info.Message
}
```

and instead put that concatenation into the new, generalized func instead

Reply:
Non-conformance is determined above via the report entry separate from the Failure message. Re-structuring to a more general `parseSpecReportMessage` would involve more than just populating `info.Message`. Maybe I'm missing your intention - perhaps you could provide a code snippet?

Reply:
@jackfrancis Does my response ☝️ satisfy your comment?